### PR TITLE
reworked Overlay

### PR DESCRIPTION
**Pull request by dominikg**
new options to position overlay to trigger element with relative values like 'top,center,left,bottom', percentages or pixels.
added configurability via appearance ( if PopoverManager finds an appearance of form 'viewmanagerid-overlay' it sets it on the viewmanagers overlay.

This changes the api of Overlay, applications that use it may require slight changes.

### CODE REVIEW

File: unify/framework/source/class/unify/ui/container/Overlay.js [hunk 1]
```diff
@@ -82,88 +100,174 @@ qx.Class.define("unify.ui.container.Overlay", {
...
-          left: left - posHint.left,
-          top: top - posHint.top
-        });
+    calculateArrowPosition : function(height,width){
+      var GAP = 6;//TODO use border width +1?
+      var arrow = this.getChildControl("arrow");
+      var arrowLeft=0;
+      var arrowTop=0;
+
+      if(arrow)
```
Comment (fastner) [inline unify/framework/source/class/unify/ui/container/Overlay.js]: Block is missing

File: unify/framework/source/class/unify/ui/container/Overlay.js [hunk 2]
```diff
@@ -34,34 +34,52 @@ qx.Class.define("unify.ui.container.Overlay", {
...
   },
+
   
   /**
-   * @param arrowDirection {String} Direction of arrow (left, top, right, bottom)
-   * @param arrowAlignment {String} Alignment of arrow on container (top, center, bottom or left, center, right)
+   * @param noArrow {Boolean?} set to true if overlay should not have an arrow element
    */
-  construct : function(arrowDirection, arrowAlignment) {
+  construct : function(noArrow) {
```
Comment (fastner) [inline unify/framework/source/class/unify/ui/container/Overlay.js:74]: noArrow seems a bit "dirty"

File: unify/framework/source/class/unify/ui/container/Overlay.js [hunk 3]
```diff
@@ -82,88 +100,174 @@ qx.Class.define("unify.ui.container.Overlay", {
...
+     * @param relativePos {Object} a map containing relative position values as keys x and y e.g. {x:"center",y:"bottom"}
+     */
+    __resolveRelative: function(elemPos,relativePos){
+      return {
+        top: (elemPos.top||0)+this.__toPixelValue(elemPos.height,relativePos.x),
+        left:(elemPos.left||0)+this.__toPixelValue(elemPos.width,relativePos.y)
+      }
+    },
+
+    __toPixelValue : function(baseSize,value){
```
Comment (fastner) [inline unify/framework/source/class/unify/ui/container/Overlay.js]: No comments

File: unify/framework/source/class/unify/ui/container/Overlay.js (hunk 1 above)
Comment (dominikg) [inline unify/framework/source/class/unify/ui/container/Overlay.js]: done

File: unify/framework/source/class/unify/ui/container/Overlay.js (hunk 2 above)
Comment (dominikg) [inline unify/framework/source/class/unify/ui/container/Overlay.js:74]: it is, but it saves creating an arrow element in the constructor and removing it via setHasArrow(false) directly afterwards

**Pull request merged**